### PR TITLE
Retry stateproof request in js monitor check

### LINKED ISSUE #5057: False positive stateproof API alert
### Description

We are getting occasional alerts on the stateproof REST API for mainnet-public. They resolve automatically after they retry 10 minutes later. Investigation reveals that they are caused by a race condition in the monitor.

### Steps to reproduce

Run monitor against mainnet-public then the following scenario will occur in `stateproof_tests.js`:

* `/api/v1/transactions?limit=1&result=success` calls mainnet-na cluster to get latest successful transaction
* `/api/v1/transactions/{txid-from-above}/stateproof?nonce=0&scheduled=false` calls mainnet-eu cluster

Clusters might process record files at slightly different times so one cluster might not have the latest txid retrieved from the other cluster. It's a race condition because the APIs are eventually consistent across multiple clusters.

### Additional context

```
[#211785: An uptime check on mirrornode mirrornode-production-health-check is failing. | Violation started: Dec 16, 2022 at 3:39PM UTC (less than 1 sec ago) | Policy: mainnet-public uptime failure | Condition: Failure of uptime check_id mainnet-public) by stackdriver
```

### Hedera network

mainnet

### Version

0.69.2

### Operating system

None

### PR DESCRIPTION
**Pull request by xin-hedera**
**Description**:

This PR adds retry to js monitor stateproof check to improve resiliency for envs with possible eventual data consistency issue.

**Related issue(s)**:

Fixes #5057 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-rest/monitoring/monitor_apis/stateproof_tests.js [hunk 1]
```diff
@@ -60,24 +64,38 @@ const checkStateproofForValidTransaction = async (server) => {
...
+      minTimeout: retry.backoff,
+      retries: retry.maxAttempts - 1,
+    });
+
+    retryOperation.attempt(async (current) => {
+      const stateproof = await getAPIResponse(url);
+      const result = checkRunner.run(stateproof);
+      if (retryOperation.retry(!result.passed)) {
+        // retry when not passed and the max number of retries haven't been reached
+        logger.warn(`Retry stateproof "${url}" after attempt #${current}`);
```
Comment (steven-sheehy) [inline hedera-mirror-rest/monitoring/monitor_apis/stateproof_tests.js:87]: Can we print what the error was that caused the retry?

Comment (xin-hedera) [reply hedera-mirror-rest/monitoring/monitor_apis/stateproof_tests.js]: done

File: hedera-mirror-rest/monitoring/monitor_apis/stateproof_tests.js [hunk 2]
```diff
@@ -60,24 +64,38 @@ const checkStateproofForValidTransaction = async (server) => {
...
+      minTimeout: retry.backoff,
+      retries: retry.maxAttempts - 1,
+    });
+
+    retryOperation.attempt(async (current) => {
+      const stateproof = await getAPIResponse(url);
+      const result = checkRunner.run(stateproof);
+      if (retryOperation.retry(!result.passed)) {
+        // retry when not passed and the max number of retries haven't been reached
+        logger.warn(`Retry after attempt #${current} failed: ${result.message}`);
```
Comment (steven-sheehy) [inline hedera-mirror-rest/monitoring/monitor_apis/stateproof_tests.js:87]: Missed this first time: We generally print out `server.run` and `server.name` due to the concurrent nature of the tests. Also should print out the url or path that was retried unless that's included in the `result.message`.

**Review by steven-sheehy — APPROVED** [review]
LGTM. Just another logging nit.

File: hedera-mirror-rest/monitoring/monitor_apis/stateproof_tests.js (hunk 2 above)
Comment (steven-sheehy) [reply hedera-mirror-rest/monitoring/monitor_apis/stateproof_tests.js:87]: Sounds good

**Review by mgoelswirlds — APPROVED** [review]
LGTM

**Pull request merged**